### ISSUE TITLE
scalene stops with `SIGSEGV` when importing `numba`

### ISSUE THREAD

**Issue by jacopok** [plasma-umass/scalene#362]
**Describe the bug**
When importing [`numba`](https://numba.pydata.org/), `scalene` crashes.

**To Reproduce**
1. Install `numba`;
2. make a file, `test.py`, containing `import numba`;
3. run `scalene test.py`
4. the output is `Scalene error: received signal SIGSEGV`.

(while running `python test.py` works with no output, as expected)

**Expected behavior**
The script is profiled (or maybe an error is raised for it being too short).

**Desktop (please complete the following information):**
 - OS: ubuntu 21.10
 - `numba` version: 0.55.1
 - `scalene` version: 1.5.4
 - `python` version: 3.9.9 or 3.10.1 (same behavior)

**Comment by sternj** [comment]
Unable to replicate on my local environment-- what are your system resources? In particular, how much RAM is available to your Python interpreter, and are you running in any sort of containerized or virtualized environment?

**Comment by sternj** [comment]
(I'm standing up a VM with the exact specified environment to try and replicate)

**Comment by jacopok** [comment]
Yes, I should have mentioned this as well, sorry: I am running this inside a `pyenv` virtual environment.

**Comment by sternj** [comment]
Yeah I am running in pyenv too-- like, are you running in a `pyenv-virtualenv` virtual environment, or just in Pyenv?

**Comment by jacopok** [comment]
I believe I am in the former case: for example, when I run `pyenv virtualenvs` the one I am working in is selected, which is the same syntax as the [pyenv-virtualenv](https://github.com/pyenv/pyenv-virtualenv) README.

I will try to make a clean one to see if the problem still occurs.

**Comment by sternj** [comment]
Would you mind joining the [Scalene slack](https://join.slack.com/t/scaleneprofil-jge3234/shared_invite/zt-110vzrdck-xJh5d4gHnp5vKXIjYD3Uwg) and messaging me (Sam Stern)? I've been hunting down an issue similar to this one and could use any help to replicate this consistently

**Comment by sternj** [comment]
Ok, replicated on a VM running Ubuntu 20.10 with Python 3.9.9

**Comment by jacopok** [comment]
Sure! This seems like an interesting problem to tackle.

**Comment by jacopok** [comment]
With edc23d55c5e355c1578b16c1b0795fb48065fa11 the issue is fixed on my end!
